circuit-overlay/src/commonMain/kotlin/com/slack/circuit/overlay/ContentWithOverlays.kt

```diff
@@ -31,7 +33,13 @@ public fun ContentWithOverlays(
   content: @Composable () -> Unit
 ) {
   val overlayHostData by rememberUpdatedState(overlayHost.currentOverlayData)
-  CompositionLocalProvider(LocalOverlayHost provides overlayHost) {
+  val overlayState by remember {
+    derivedStateOf { overlayHostData?.let { OverlayState.SHOWING } ?: OverlayState.HIDDEN }
+  }
+  CompositionLocalProvider(
+    LocalOverlayHost provides overlayHost,
```

Comment:
We shouldn't need to re-provide the `LocalOverlayHost` here

Reply:
We're not guaranteed to have a host here currently, since this is the primary entry point. This is what we were doing before too. Am I missing something?

Reply:
Ohh nope, just mis-read the above lines 🤦🏻